Trappist/src/Promact.Trappist.Test/Reports/ReportsRepositoryTest.cs

```diff
@@ -137,5 +142,191 @@ private TestAttendees TestAttendeeReport(int id)
             await _testRepository.CreateTestAsync(test, applicationUser.Id);
             return test;
         }
+
+        /// <summary>
+        /// Gets the details of a test attendee along with his marks and test logs by his Id
+        /// </summary>
+        [Fact]
+        public async Task GetTestAttendeeDetailsByIdAsync()
+        {
+            var test = CreateTest("Mathematics");
+            await _testRepository.CreateTestAsync(test, "1");
+            var testAttendee = CreateTestAttendee(test.Id);
+            await _testConductRepository.RegisterTestAttendeesAsync(testAttendee, "Added Successfully");
+            var testAttendeeObject = await _reportRepository.GetTestAttendeeDetailsByIdAsync(testAttendee.Id);
+            Assert.NotNull(testAttendeeObject);
+        }
+
+        /// <summary>
+        /// Gets all the questions present in a test by test Id
+        /// </summary>
+        [Fact]
+        public async Task GetTestQuestions()
+        {
+            var test = CreateTest("Mathematics");
+            await _testRepository.CreateTestAsync(test, "4");
+            var category = CreateCategory("Maths");
+            var questionToCreate1 = "Question1";
+            var question1 = CreateSingleAnswerQuestion(category, questionToCreate1);
+
+            var questionToCreate2 = "Question2";
+            var question2 = CreateSingleAnswerQuestion(category, questionToCreate2);
+            var testCategoryObject = new TestCategory()
+            {
+                CategoryId = category.Id,
+                Test = test,
+                TestId = test.Id,
+                Category = category
+            };
+            _trappistDbContext.TestCategory.Add(testCategoryObject);
+            await _trappistDbContext.SaveChangesAsync();
+            var testQuestionObject1 = new TestQuestion()
+            {
+                QuestionId = question1.Id,
+                TestId = test.Id,
+                Test = test,
+
+            };
+            var testQuestionObject2 = new TestQuestion()
+            {
+                QuestionId = question2.Id,
+                TestId = test.Id,
+                Test = test,
+            };
+            var testQuestionList = new List<TestQuestion>();
+            testQuestionList.Add(testQuestionObject1);
+            testQuestionList.Add(testQuestionObject2);
+            await _trappistDbContext.TestQuestion.AddRangeAsync(testQuestionList);
+            await _trappistDbContext.SaveChangesAsync();
+            var testAttendee = CreateTestAttendee(test.Id);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused 'testAttendee' local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)